File: backend/app/services/extract_service.py
import json

from app.services.gemini_service import client, MODEL_NAME


class ExtractService:

    @staticmethod
    def extract(conversation: str):

        prompt = f"""
You are an AI legal assistant.

Extract the following information from the conversation.

Return ONLY valid JSON.

Schema:

{{
    "consumer_name": "",
    "seller": "",
    "product": "",
    "issue": "",
    "facts": "",
    "relief": ""
}}

Conversation:

{conversation}
"""

        response = client.models.generate_content(
            model=MODEL_NAME,
            contents=prompt,
        )

        text = response.text.strip()

        # Remove markdown if Gemini wraps JSON
        text = text.replace("```json", "").replace("```", "").strip()

        return json.loads(text)

File: frontend/services/extractService.ts
import api from "@/lib/api";

export interface ExtractRequest {
  conversation: string;
}

export interface ExtractResponse {
  consumer_name: string;
  seller: string;
  product: string;
  issue: string;
  facts: string;
  relief: string;
}

export async function extractCase(
  conversation: string
): Promise<ExtractResponse> {
  const response = await api.post("/extract-case/", {
    conversation,
  });

  return response.data;
}